spatio_temporal_voxel_layer/src/spatio_temporal_voxel_layer.cpp

```diff
@@ -533,17 +549,43 @@ bool SpatioTemporalVoxelLayer::updateFootprint(
   double * min_y, double * max_x, double * max_y)
 /*****************************************************************************/
 {
-  // updates layer costmap to include footprint for clearing in voxel grid
+  // Updates layer costmap to include footprint for clearing in voxel grid
   if (!_update_footprint_enabled) {
     return false;
   }
-  nav2_costmap_2d::transformFootprint(
-    robot_x, robot_y, robot_yaw,
-    getFootprint(), _transformed_footprint);
-  for (unsigned int i = 0; i < _transformed_footprint.size(); i++) {
-    touch(
-      _transformed_footprint[i].x, _transformed_footprint[i].y,
-      min_x, min_y, max_x, max_y);
+
+  if (!_footprint_projection_enabled) {
+    // Simple 2D transformation
+    nav2_costmap_2d::transformFootprint(
+      robot_x, robot_y, robot_yaw,
+      getFootprint(), _transformed_footprint);
+    for (unsigned int i = 0; i < _transformed_footprint.size(); i++) {
+      touch(
+        _transformed_footprint[i].x, _transformed_footprint[i].y,
+        min_x, min_y, max_x, max_y);
+    }
+  } else {
+    // Using tf2 for 3d rotation to provide accurate projection of the footprint
+    try {
+      geometry_msgs::msg::TransformStamped tf_footprint_stamped =
+        tf_->lookupTransform(
+          _global_frame, _footprint_frame,
+          rclcpp::Time(0)); 
+      
+      _transformed_footprint = getFootprint();
+      for (unsigned int i = 0; i < _transformed_footprint.size(); i++) {
+        tf2::doTransform(_transformed_footprint[i], _transformed_footprint[i], tf_footprint_stamped);       
```

Comment:
You're aware you're updating the stored member variable when you do this, right? Its not just updating it for local use in this method

Reply:
I am, original method also does this in order for the `_transformed_footprint` to be used in `updateCosts` method.